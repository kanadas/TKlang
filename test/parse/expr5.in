(\a b -> if a + b == /*tralalalaal*/0 then "Zero" else a + b)
