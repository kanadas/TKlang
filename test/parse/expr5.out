parse/expr5.in

Parse Successful!

[Abstract Syntax]

ELambda [Ident "a",Ident "b"] (EIf (ERel (EAdd (EIdent (Ident "a")) (EIdent (Ident "b"))) (RelOp "==") (EInt 0)) (EString "Zero") (EAdd (EIdent (Ident "a")) (EIdent (Ident "b"))))

[Linearized tree]

(\ a b -> if a + b == 0 then "Zero" else a + b)
